src/main/java/com/microsoft/sqlserver/jdbc/SQLServerCallableStatement.java

```diff
@@ -1348,13 +1348,17 @@ private int findColumn(String columnName) throws SQLServerException {
 
         }
 
-        int l = 0;
+        int parameterNamesSize = 0;
+
         if (null != parameterNames) {
-            l = parameterNames.size();
+            parameterNamesSize = parameterNames.size();
         }
-        if (l == 0) { // Server didn't return anything, user might not have access
-            map.putIfAbsent(columnName, ai.incrementAndGet());
-            return map.get(columnName);// attempting to look up the first column will return no access exception
+
+        // @RETURN_VALUE will always be in the parameterNames map, so parameterNamesSize will always be at least of size 1.
+        // If the server didn't return anything (eg. the param names for the sproc), user might not have access.
+        // So, parameterNamesSize must be of size 1.
+        if (parameterNamesSize == 1) {
```

Comment:
```suggestion
        if (null != parameterNames && parameterNames.size() == 1) {
```
I think we can remove 1351 - 1355. parameterNamesSize only has 1 use, to see if its 1, so we don't really need it to be defined, and I think we can consolidate it use to just a line, without it looking too ugly.